src/TwigComponent/src/HasAttributesTrait.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\LiveComponent\Attribute\PostHydrate;
+use Symfony\UX\TwigComponent\Attribute\PostMount;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @experimental
+ */
+trait HasAttributesTrait
+{
+    #[LiveProp(hydrateWith: 'hydrateAttributes', dehydrateWith: 'dehydrateAttributes')]
+    public ComponentAttributes $attributes;
+
+    public function setAttributes(array|ComponentAttributes $attributes): void
+    {
+        $this->attributes = \is_array($attributes) ? new ComponentAttributes($attributes) : $attributes;
+    }
+
+    #[PostMount(priority: -1000)]
+    #[PostHydrate]
```

Comment:
Why also on PostHydrate? This method may deserve some phpdoc explanation

Reply:
Heh, this took me while to figure out... When dehydrating a component with empty attributes, there is no attribute key in the dehydrated array (it's filtered out). In this case, when re-hydrating, the `hydrateAttributes()` is never called so the `$attributes` property is never instantiated. The `PostHydrate` ensures this method is always called after hydrating.

Reply:
> When dehydrating a component with empty attributes, there is no attribute key in the dehydrated array (it's filtered out).

What filters it out exactly? This still feels like maybe we're covering up something that doesn't work quite right...

Reply:
Couldn't recreate the problem so I removed.